src/services/completions.ts

```diff
@@ -1650,8 +1654,9 @@ function getEntryForMemberCompletion(
     return { insertText, isSnippet, importAdder, replacementSpan };
 }
 
-function getPresentModifiers(contextToken: Node | undefined): { modifiers: ModifierFlags, span?: TextSpan } {
-    if (!contextToken) {
+function getPresentModifiers(contextToken: Node | undefined, position: number): { modifiers: ModifierFlags, span?: TextSpan } {
+    if (!contextToken
+        || getLineAndCharacterOfPosition(contextToken.getSourceFile(), position).line > getLineAndCharacterOfPosition(contextToken.getSourceFile(), contextToken.getEnd()).line) {

```

Comment:
this line is still a little too long -- it wraps on on a 1440px-wide screen at least.

Reply:
You are also requesting the `SourceFile` several times, and that never gets cached across nodes. Fetch it once and pass the same value through (or use the available `sourceFile` that was declared above?)